bin/config.sh

```diff
@@ -55,4 +55,9 @@ function Upload {
     mc cp $file spaces/edm-publishing/ceqr-app-data-staging/$1/$2/$name
   done
   wait
+}
+
+function Publish {
+  mc rm -r --force spaces/edm-publishing/ceqr-app-data/$1/$2
+  mc cp -r spaces/edm-publishing/ceqr-app-data-staging/$1/$2 spaces/edm-publishing/ceqr-app-data/$1/
 }
```

Comment:
```bash
function Publish {
  echo "publishing ceqr-app-data-staging/$1/$2"
  mc rm -r --force spaces/edm-publishing/ceqr-app-data/$1/$2
  mc cp -r spaces/edm-publishing/ceqr-app-data-staging/$1/$2 /tmp/ceqr-app-data-staging/$1/$2
  mc cp -r /tmp/ceqr-app-data-staging/$1/$2 spaces/edm-publishing/ceqr-app-data/$1/$2

  echo "promote $1/$2 to latest"
  mc rm -r --force spaces/edm-publishing/ceqr-app-data/$1/latest 
  mc cp -r /tmp/ceqr-app-data-staging/$1/$2 spaces/edm-publishing/ceqr-app-data/$1/latest
  rm -rf /tmp/ceqr-app-data-staging/$1/$2
}
register 'publish' 'recipe' '{ recipe name } { version name } e.g. nysdot_aadt 2020-07-16' Publish

function List {
  if [[ "$2" == "production" ]]
  then
    echo "
    listing ceqr-app-data/$1
    "
    mc ls spaces/edm-publishing/ceqr-app-data/$1
  else
    echo "
    listing ceqr-app-data-staging/$1
    "
    mc ls spaces/edm-publishing/ceqr-app-data-staging/$1
  fi
}
register 'list' 'recipe' '{ recipe name } { staging or production}' List
```

Maybe publish should be a manual process otherwise production = staging
the two command above will list versions in buckets e.g. 
```bash
# default listing staging
./ceqr list recipe nysdot_aadt 

# or list versions in production
./ceqr list recipe nysdot_aadt production
```

we can then pick the version we want to promote to production e.g.
```bash 
./ceqr publish recipe nysdot_aadt 2020-07-16

# or syncing everything between staging and production (dangerous)
./ceqr publish recipe nysdot_aadt
```

Reply:
I noticed some performance issues doing cross bucket copy, there were also rate limits.
so I added a local IO, which made things faster
```bash
  mc cp -r spaces/edm-publishing/ceqr-app-data-staging/$1/$2 /tmp/ceqr-app-data-staging/$1/$2
```